packages/store-sdk/src/cart/Optimistic.tsx

```diff
@@ -0,0 +1,68 @@
+import PQueue from 'p-queue'
+import React, { createContext, useEffect, useMemo, useState } from 'react'
+import type { FC } from 'react'
+
+import { useContext } from '../utils/useContext'
+import { Context as CartContext } from './Cart'
+import type { ContextValue as CartContextValue } from './Cart'
+
+export type Cart = Pick<CartContextValue, 'id' | 'items'>
```

Comment:
Type it as `Cart<T extends Item = Item> = Pick<CartContextValue<T>, 'id' | 'items'>` to make the types work with the Provider props.